mountpoint-s3-client/src/object_client.rs

```diff
@@ -103,6 +103,14 @@ pub trait ObjectClient {
         key: &str,
     ) -> ObjectClientResult<DeleteObjectResult, DeleteObjectError, Self::ClientError>;
 
+    async fn copy_object(
```

Comment:
Should we add a `CopyObjectParams` argument? Even if it starts empty in this PR, it would help to avoid future breaking changes.

Reply:
ack, I will add CopyObjectParams, we can extend that as and when we have new use cases.